Products/PortalTransforms/transforms/safe_html.py

```diff
@@ -2700,25 +2544,48 @@ def convert(self, orig, data, **kwargs):
         # we need a disable option
         if self.config.get('disable_transform'):
             data.setData(orig)
-            return data
-
-        for repeat in range(2):
-            try:
-                safe = scrubHTML(
-                    bodyfinder(orig),
-                    valid=self.config.get('valid_tags', {}),
-                    nasty=self.config.get('nasty_tags', {}),
-                    remove_javascript=self.config.get(
-                        'remove_javascript', True),
-                    raise_error=False)
-            except IllegalHTML as inst:
-                data.setData(msg_pat % ("Error", str(inst)))
-                break
-            else:
-                data.setData(safe)
-                orig = safe
+        else:
+            safe_html = self.scrub_html(orig)
+            data.setData(safe_html)
         return data
 
+    def scrub_html(self, orig):
+        # append html tag to create a dummy parent for the tree
+        html_parser = html.HTMLParser(encoding='utf-8')
+        if '<html' in orig.lower():
+            # full html
+            tree = html.fromstring(orig, parser=html_parser)
+            strip_outer = bodyfinder
+        else:
+            # partial html (i.e. coming from WYSIWYG editor)
+            tree = html.fragment_fromstring(orig, create_parent=True, parser=html_parser)
+
+            def strip_outer(s):
+                return s[5:-6]
+
+        for elem in tree.iter(etree.Element):
+            if elem is not None:
+                for attrib, value in elem.attrib.items():
+                    if hasScript(value):
+                        del elem.attrib[attrib]
+
+        cleaner = Cleaner(kill_tags=self.config['nasty_tags'],
+                          page_structure=False,
+                          safe_attrs_only=False,
+                          embedded=False,
+                          remove_unknown_tags=True,
+                          meta=False,
+                          javascript=bool('script' in self.config['nasty_tags']),
+                          scripts=bool('script' in self.config['nasty_tags']),
```

Comment:
> 5.9. Comparisons
> ...
> Comparisons yield boolean values: True or False.
> ...
> The operators in and not in test for collection membership. x in s evaluates to true if x is a member of the collection s, and false otherwise. x not in s returns the negation of x in s.

https://docs.python.org/2/reference/expressions.html#not-in

IOW: boolean casting is superfluous, except one deals with very broken code, which is not the case here.